evap/staff/importers.py

```diff
@@ -247,6 +248,14 @@ def check_course_data_correctness(self, semester):
             if not Degree.objects.filter(name_de=degree_name).exists():
                 self.errors.append(_("Error: The degree \"{}\" does not exist yet. Please manually create it first.").format(degree_name))
 
+        course_type_names = set()
+        for course_data in self.courses.values():
+            course_type_names.add(course_data.type_name)
```

Comment:
you could do `course_type_names = set(course_data.type_name for course_data in self.courses.values())`